### PR TITLE
Add jobs and role for scanning containers

### PR DESCRIPTION
**Pull request by mattymo**

### CODE REVIEW

File: roles/contrail-scan-container/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+- name: Ensure jq is installed
+  package:
+    name: jq
+    state: present
+
+- name: Scan docker registry catalog for matching images
+  shell: >-
+    curl -s http://{{ nexus.fqdn }}:{{ nexus.registry.tungsten_ci }}/v2/_catalog |
```
Comment (alexey-mr) [inline roles/contrail-scan-container/tasks/main.yaml]: would it be more readable if have separate bash script?

File: roles/contrail-scan-container/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  register: scan_raw
+  with_items: "{{ docker_images_to_scan }}"
+  failed_when: false
+
+- name: Parse results to set max_score per image
+  set_fact:
+    scan_results: >-
+      {
+      {% for image, report in dict(docker_images_to_scan | zip(scan_raw.results | map(attribute='stdout'))).iteritems() %}
+      {%   set max_score = (report |from_json).vulnerability_summary.max_score %}
```
Comment (alexey-mr) [inline roles/contrail-scan-container/tasks/main.yaml]: do we have both vendor and cvss scores here?

File: roles/contrail-scan-container/defaults/main.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+nexus:
```
Comment (Andrey-mp) [inline roles/contrail-scan-container/defaults/main.yaml]: do you really need this file? all of these variables are defined in stite-variables... maybe except max_security_threshold

File: zuul.d/base-jobs.yaml [hunk 4]
```diff
@@ -165,6 +165,18 @@
...
+    name: contrail-security-scan-base
+    parent: contrail-sanity-base
+    description: |
+      Base job for using AquaScan to check for security vulnerabilities
+      in built containers.
+    pre-run: playbooks/contrail-security-scan-base/pre.yaml
+    vars:
+      aquasec_version: "4.2"
+      max_security_threshold: "7.0"
+    nodeset: ubuntu-systest-c-multinode
```
Comment (Andrey-mp) [inline zuul.d/base-jobs.yaml]: do you really need multinode environment?

File: zuul.d/contrail-jobs.yaml [hunk 5]
```diff
@@ -192,6 +192,14 @@
       test_target: ci_sanity
     timeout: 10800
 
+- job:
+    name: contrail-security-scan-rhel-queens
+    parent: contrail-security-scan-base
+    run: playbooks/contrail-security-scan-rhel-queens/run.yaml
+    vars:
+      openstack_version: queens
+    timeout: 10800
```
Comment (Andrey-mp) [inline zuul.d/contrail-jobs.yaml]: is it really so long?

File: roles/contrail-scan-container/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  with_items: "{{ docker_images_to_scan }}"
+  failed_when: false
+
+- name: Scan images
+  shell: >-
+    docker run
+    -v /var/run/docker.sock:/var/run/docker.sock
+    {{ nexus.fqdn }}:{{ nexus.registry.aquasec }}/scanner:4.2
+    scan
+    -H http://{{ nexus.fqdn }}:8080
```
Comment (Andrey-mp) [inline roles/contrail-scan-container/tasks/main.yaml]: it uses nexus:8080. is it for logs upload?

**Comment by Andrey-mp** [comment]
- why do you need 3 jinja templates? they are used for build mostly.
- i don't see 2 new playbooks

File: roles/contrail-scan-container/defaults/main.yaml (hunk 3 above)
Comment (mattymo) [reply roles/contrail-scan-container/defaults/main.yaml]: correct. I'll just set the max_security_threshold then

File: roles/contrail-scan-container/tasks/main.yaml (hunk 1 above)
Comment (mattymo) [reply roles/contrail-scan-container/tasks/main.yaml]: It won't change things so much. It's broken up at least. The difficult logic is the two jq commands

File: roles/contrail-scan-container/tasks/main.yaml (hunk 6 above)
Comment (mattymo) [reply roles/contrail-scan-container/tasks/main.yaml]: here are the ports on nexus host:
8080 for aquascan
80 for zuul logs
8081 for nexus

File: roles/contrail-scan-container/tasks/main.yaml (hunk 2 above)
Comment (mattymo) [reply roles/contrail-scan-container/tasks/main.yaml]: they exist. this is just the summary. I'll send you a complete report to look at

File: roles/contrail-scan-container/templates/common.env.j2 [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{% if ansible_distribution == 'RedHat' %}
```
Comment (mattymo) [inline roles/contrail-scan-container/templates/common.env.j2]: this file needs to be deleted

File: roles/contrail-scan-container/templates/google-chrome.repo.j2 [hunk 8]
```diff
@@ -0,0 +1,6 @@
+[google-chrome]
```
Comment (mattymo) [inline roles/contrail-scan-container/templates/google-chrome.repo.j2]: this file needs to be deleted

File: roles/contrail-scan-container/templates/tpc.repo.template.j2 [hunk 9]
```diff
@@ -0,0 +1,5 @@
+[tpc]
```
Comment (mattymo) [inline roles/contrail-scan-container/templates/tpc.repo.template.j2]: this file needs to be deleted

File: zuul.d/base-jobs.yaml (hunk 4 above)
Comment (mattymo) [reply zuul.d/base-jobs.yaml]: nope. any node set is fine

File: zuul.d/contrail-jobs.yaml (hunk 5 above)
Comment (mattymo) [reply zuul.d/contrail-jobs.yaml]: the others have such a timeout... this job could take an hour in some extreme case

File: roles/contrail-scan-container/tasks/main.yaml (hunk 2 above)
Comment (mattymo) [reply roles/contrail-scan-container/tasks/main.yaml]: We discussed and decided to find the largest nvd_score and vendor_score and use that

**Pull request merged**